File: edp-frontend/src/app/career-package/submission/models/submission-snapshot-response.dto.ts
import { User } from "../../../user/models/user.model";
import { SubmissionSectionDto } from "./section-tag-snapshot-response.dto";


export interface SubmissionSnapshotResponseDto {
  submissionId: number;
  user: User;
  department: string;
  position: string;
  status: string;
  comment: string;
  submittedAt: string; 
  reviewedAt: string; 
  sections: SubmissionSectionDto[];
}

File: career-package-service/src/main/java/com/edp/careerpackage/model/submissionsnapshot/SubmissionSnapshotResponseDto.java
package com.edp.careerpackage.model.submissionsnapshot;

import com.edp.shared.client.auth.model.UserProfileDto;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Builder
public class SubmissionSnapshotResponseDto {
    private Long submissionId;
    private UserProfileDto user;
    private String department;
    private String position;
    private String status;
    private String comment;
    private LocalDateTime submittedAt;
    private LocalDateTime reviewedAt;
    private List<SubmissionSectionSnapshotResponseDto> sections;
}